packages/emitter-framework/src/python/components/class-declaration/class-declaration.test.tsx

```diff
@@ -0,0 +1,855 @@
+    const { program, Person } = await Tester.compile(t.code`
+      model ${t.model("Person")} is Record<string> {
+        name: string;
+      }
+    `);
+
+    expect(() => {
+      expect(getOutput(program, [<ClassDeclaration type={Person} />])).toRenderTo("");
+    }).toThrow(/Models with additional properties \(Record\[…\]\) are not supported/);
+  });
+
+  it("throws error for model extends Record<string>", async () => {
+    const { program, Person } = await Tester.compile(t.code`
+      model ${t.model("Person")} extends Record<string> {
+        name: string;
+      }
+    `);
+
+    expect(() => {
+      expect(getOutput(program, [<ClassDeclaration type={Person} />])).toRenderTo("");
+    }).toThrow(/Models with additional properties \(Record\[…\]\) are not supported/);
+  });
+
+  it("throws error for model with ...Record<string>", async () => {
+    const { program, Person } = await Tester.compile(t.code`
+      model ${t.model("Person")} {
+        age: int32;
+        ...Record<string>;
+      }
+    `);
+
+    expect(() => {
+      expect(getOutput(program, [<ClassDeclaration type={Person} />])).toRenderTo("");
+    }).toThrow(/Models with additional properties \(Record\[…\]\) are not supported/);
+  });
+
+  it("creates a class from a model that 'is' an array ", async () => {
+    const { program, Foo } = await Tester.compile(t.code`
+      model ${t.model("Foo")} is Array<string>;
+    `);
+
+    expect(getOutput(program, [<ClassDeclaration type={Foo} />])).toRenderTo(
+      `
+      from dataclasses import dataclass
+
+      @dataclass
+      class Foo(list[str]):
+        pass
+
+    `,
+    );
+  });
+
+  it("handles a type reference to a union variant in a class property", async () => {
+    const { program, Color, Widget } = await Tester.compile(t.code`
+      union ${t.union("Color")} {
+        red: "RED",
+        blue: "BLUE"
+      }
+  
+      model ${t.model("Widget")} {
+        id: string = "123";
+        weight: int32 = 100;
+        color: Color.blue
+      }
+      `);
+
+    expect(
+      getOutput(program, [<EnumDeclaration type={Color} />, <ClassDeclaration type={Widget} />]),
+    ).toRenderTo(
+      `
+      from dataclasses import dataclass
+      from enum import StrEnum
+      from typing import Literal
+
+      class Color(StrEnum):
+        RED = "RED"
+        BLUE = "BLUE"
+
+
+      @dataclass
+      class Widget:
+        id: str = "123"
+        weight: int = 100
+        color: Literal[Color.BLUE]
+
+      `,
+    );
+  });
+
+  it("renders an empty class based on a model with a never-typed member", async () => {
+    const { program, Widget } = await Tester.compile(t.code`
+    model ${t.model("Widget")} {
+      property: never;
+    }
+    `);
+
+    expect(getOutput(program, [<ClassDeclaration type={Widget} />])).toRenderTo(`
+      from dataclasses import dataclass
+
+      @dataclass
+      class Widget:
+        pass
```

Comment:
Should we declare `property` to have [the `Never` type](https://typing.python.org/en/latest/spec/special-types.html#never)?